src/main/java/com/puppycrawl/tools/checkstyle/JavadocPropertiesGenerator.java

```diff
@@ -103,36 +101,42 @@ public static void main(String... args)
      * @param inputFile .java file
      * @param outputFile .properties file
      * @throws CheckstyleException if a javadoc comment can not be parsed
-     * @throws IOException if there is a problem with files access
+     * @noinspection IOResourceOpenedButNotSafelyClosed
      */
     private static void writePropertiesFile(String inputFile, String outputFile)
-            throws CheckstyleException, IOException {
-        final FileOutputStream fos = new FileOutputStream(outputFile);
-        final Writer osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
-        final PrintWriter writer = new PrintWriter(osw, false);
-
+            throws CheckstyleException {
+        PrintWriter writer = null;
         try {
-            for (DetailAST top = JavaParser.parseFile(new File(inputFile),
-                    JavaParser.Options.WITH_COMMENTS); top != null; top = top.getNextSibling()) {
-                if (top.getType() != TokenTypes.CLASS_DEF) {
-                    continue;
-                }
-                // OBJBLOCK is always the last child of CLASS_DEF.
-                final DetailAST objBlock = top.getLastChild();
-                for (DetailAST ast = objBlock.getFirstChild(); ast != null;
-                        ast = ast.getNextSibling()) {
-                    if (isPublicStaticFinalIntField(ast)) {
-                        final DetailAST modifiers = ast.findFirstToken(TokenTypes.MODIFIERS);
-                        final String firstJavadocSentence = getFirstJavadocSentence(modifiers);
-                        if (firstJavadocSentence != null) {
-                            writer.println(getName(ast) + "=" + firstJavadocSentence.trim());
-                        }
+            final FileOutputStream fos = new FileOutputStream(outputFile);
+            final Writer osw = new OutputStreamWriter(fos, StandardCharsets.UTF_8);
+            writer = new PrintWriter(osw, false);
+            DetailAST top = JavaParser.parseFile(new File(inputFile),
+                JavaParser.Options.WITH_COMMENTS);
+
+            // Skip the package definition and all imports.
```

Comment:
Each time engineer try put comment on block of code , he should stop and think how to avoid it by embeding words of comment in code: methods names, variable names, .....

Please create method that will give you objblock.

Reply:
done